### ISSUE TITLE
Parallelization in runtests.jl using ReTestItems and CpuId will not work for Apple M1 Pro CPU

### ISSUE THREAD

**Issue by abpolym** [ReactiveBayes/ReactiveMP.jl#434]
I [noticed](https://github.com/m-j-w/CpuId.jl/issues/64) that CpuId yields default values when running `make test` on an Apple M1 Pro CPU:

```bash
$ julia --project=. -e "using CpuId; println(cpucores())"
0
```

This means that `CpuId`'s `cputhreads()` and `cpucores()` will return the default value `0` and consequently `ReTestItems`' `runtests` will run with only `1` thread and  `1` core.

To be able to run tests in parallel on an M1 chip, proposed solutions are

- add some special case for Apple M1 chips in `runtests.jl` before the issue in CpuId is fixed
- locate a different package that also returns threads/cores information for M1 chips 

The last commit to the `CpuId` package is over one year old, so the package seems to be unmaintained.

**Comment by bvdmitri** [comment]
Yes, for this reason we have max(cpucores(), 1) in the source code. It would be nice to have, but I rarely execute all tests via terminal. I find it easier to execute individual/all tests via VSCode panel and it doesn't use the runtests.jl

**Comment by abpolym** [comment]
Also quoting `CpuId`'s README, `CpuId` will not work for any ARM-based machine:

`Works on Julia 1.0 and later, on Linux, Mac and Windows with Intel CPUs and AMD CPUs. Other processor types like ARM are not supported.`

**Comment by bvdmitri** [comment]
I relabeled it from bug to enhancement. Performance issues aren't really bugs, it works fine and as expected, just not as fast as it could be. It would be nice of course to enable parallel test execution and I also labeled it as "good first issue". Perhaps we can pitch this issue in Active Inference discord to see if someone is interested to contribute?

**Comment by abpolym** [comment]
You can also assign me, I think I can fix that this week easily

**Comment by abpolym** [comment]
I will fix this myself so I can work on issues that require tests
